src/merge/OffenceCodeMerger.ts

```diff
@@ -0,0 +1,70 @@
+import { OffenceCode } from "../types/OffenceCode"
+import getMatchCjsCodeFunction from "./getMatchCjsCodeFunction"
+import HomeOfficeClassifictionPriority from "./HomeOfficeClassificationPriority"
+import NotifiableToHOPriority from "./NotifiableToHOPriority"
+import OffenceCategoryPriority from "./OffenceCategoryPriority"
+import OffenceTitlePriority from "./OffenceTitlePriority"
+import RecordableOnPncPriority from "./RecordableOnPncPriority"
+
+export default class OffenceCodeMerger {
+  currentOffenceCodes: OffenceCode[]
+
+  unsupportedOffenceCodes: string[]
+
+  legacyOverrides: OffenceCode[]
+
+  homeOfficeClassification: HomeOfficeClassifictionPriority
+
+  notifiableToHo: NotifiableToHOPriority
+
+  offenceCategory: OffenceCategoryPriority
+
+  offenceTitle: OffenceTitlePriority
+
+  recordableOnPnc: RecordableOnPncPriority
+
+  constructor(
+    currentOffenceCodes: OffenceCode[],
+    unsupportedOffenceCodes: string[],
+    legacyOverrides: OffenceCode[],
+    homeOfficeClassification: HomeOfficeClassifictionPriority,
+    notifiableToHo: NotifiableToHOPriority,
+    offenceCategory: OffenceCategoryPriority,
+    offenceTitle: OffenceTitlePriority,
+    recordableOnPnc: RecordableOnPncPriority
+  ) {
+    this.currentOffenceCodes = currentOffenceCodes
+    this.unsupportedOffenceCodes = unsupportedOffenceCodes
+    this.legacyOverrides = legacyOverrides
+    this.homeOfficeClassification = homeOfficeClassification
+    this.notifiableToHo = notifiableToHo
+    this.offenceCategory = offenceCategory
+    this.offenceTitle = offenceTitle
+    this.recordableOnPnc = recordableOnPnc
+  }
+
+  merge(): OffenceCode[] {
+    const mergedOffenceCodes: OffenceCode[] = []
```

Comment:
I'd probably just use a .map() function here rather than looping and accumulating, but not a dealbreaker